api/src/main/resources/workbench.yaml

```diff
@@ -580,3 +599,67 @@ definitions:
       domainId:
         description: clue to determine which tables to search
         type: string
+
+  SearchRequest:
+    type: object
+    required:
+      - type
+      - searchParameters
+      - modifiers
+    properties:
+      type:
+        description: type of criteria
+        type: string
+      searchParameters:
+        description: criteria by category code
+        type: "array"
+        items:
+          $ref: "#/definitions/SearchParameter"
+      modifiers:
+        description: criteria by operation or predicate
+        type: "array"
+        items:
+          $ref: "#/definitions/Modifier"
+
+  SearchParameter:
+    type: object
+    required:
+      - code
+      - domainId
+    properties:
+      code:
+        type: string
+      domainId:
+        type: string
+
+  Modifier:
+    type: object
+    required:
+      - name
+      - operator
+      - values
+    properties:
+      name:
+        type: string
+      operator:
+        type: string
+      operands:
+        type: "array"
+        items:
+          type: string
+
+  SubjectListResponse:
+    type: object
+    required:
+      - items
+    properties:
+      items:
+        type: "array"
+        items:
+          $ref: "#/definitions/Subject"
```

Comment:
Given performance concerns, it might be better to just return an array of strings here.

Either way, use the description field to document what those strings represent.

Reply:
> Given performance concerns, it might be better to just return an array of strings here.

I agree, however, implementing this change will require changes in several different places in the front and back end - I had intended to make those changes as their own PR.